flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/DescribeCatalogOperation.java

```diff
@@ -0,0 +1,114 @@
+import org.apache.flink.table.api.internal.TableResultInternal;
+import org.apache.flink.table.catalog.CatalogDescriptor;
+import org.apache.flink.table.catalog.CommonCatalogOptions;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.table.utils.EncodingUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.flink.table.api.internal.TableResultUtils.buildTableResult;
+
+/** Operation to describe a DESCRIBE CATALOG catalog_name statement. */
+@Internal
+public class DescribeCatalogOperation implements Operation, ExecutableOperation {
+
+    private final String catalogName;
+    private final boolean isExtended;
+
+    public DescribeCatalogOperation(String catalogName, boolean isExtended) {
+        this.catalogName = catalogName;
+        this.isExtended = isExtended;
+    }
+
+    public String getCatalogName() {
+        return catalogName;
+    }
+
+    public boolean isExtended() {
+        return isExtended;
+    }
+
+    @Override
+    public String asSummaryString() {
+        Map<String, Object> params = new LinkedHashMap<>();
+        params.put("identifier", catalogName);
+        params.put("isExtended", isExtended);
+        return OperationUtils.formatWithChildren(
+                "DESCRIBE CATALOG", params, Collections.emptyList(), Operation::asSummaryString);
+    }
+
+    @Override
+    public TableResultInternal execute(Context ctx) {
+        CatalogDescriptor catalogDescriptor =
+                ctx.getCatalogManager()
+                        .getCatalogDescriptor(catalogName)
+                        .orElseThrow(
+                                () ->
+                                        new ValidationException(
+                                                String.format(
+                                                        "Cannot obtain metadata information from Catalog %s.",
+                                                        catalogName)));
+        Map<String, String> properties = catalogDescriptor.getConfiguration().toMap();
+        List<List<Object>> rows =
+                new ArrayList<>(
+                        Arrays.asList(
+                                Arrays.asList("Name", catalogName),
+                                Arrays.asList(
+                                        "Type",
+                                        properties.getOrDefault(
+                                                CommonCatalogOptions.CATALOG_TYPE.key(), "")),
+                                Arrays.asList("Comment", "") // TODO: retain for future needs
+                                ));
+        if (isExtended) {
+            rows.add(Arrays.asList("Properties", convertPropertiesToString(properties)));
+        }
+
+        return buildTableResult(
+                Arrays.asList("catalog_description_item", "catalog_description_value")
```

Comment:
I understand that Spark inspired this naming style, but considering Flink's own style, we have not introduced snake case naming convention for result display.

Take Flink's `show table` command, for example.
```sql
show tables;
+------------+
| table name |
+------------+
|   MyTable5 |
|   MyTable6 |
|    MyView5 |
|    MyView6 |
+------------+
4 rows in set
!ok
```

On the other hand, I think Spark itself still does not use this schema.

E.g. `describe catalog` command from Spark
```sql
> DESCRIBE CATALOG main;
 info_name     info_value
 ------------  ------------------------------------
 Catalog Name                                  main
      Comment           Main catalog (auto-created)
        Owner                 metastore-admin-users
 Catalog Type                               Regular

> DESCRIBE CATALOG EXTENDED main;
 info_name     info_value
 ------------  ------------------------------------
 Catalog Name                                  main
      Comment  This is a reserved catalog in Spark.
      Comment           Main catalog (auto-created)
        Owner                 metastore-admin-users
 Catalog Type                               Regular
   Created By
   Created At
   Updated By
   Updated At
```

Actually, this schema is from Spark's `describe database`
```sql
DESCRIBE DATABASE employees;
  +-------------------------+-----------------------------+
  |database_description_item|database_description_value   |
  +-------------------------+-----------------------------+
  |Database Name            |employees                    |
  |Description              |For software companies       |
  |Location                 |file:/Users/Temp/employees.db|
  +-------------------------+-----------------------------+
```

Based on the above, I suggest that we draw inspiration from Spark for the schema names and consider maintaining consistency with Flink's naming style, using 'info name' and 'info value' as the schema. WDYT?

Ref
1. https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-aux-describe-catalog.html
2. https://spark.apache.org/docs/3.0.0-preview/sql-ref-syntax-aux-describe-database.html

Reply:
reasonable enough, done :)